packages/sign-client/src/controllers/engine.ts

```diff
@@ -434,9 +433,15 @@ export class Engine extends IEngine {
         params: getSdkError("USER_DISCONNECTED"),
         throwOnFailedPublish: true,
       });
-      await this.deleteSession(topic);
-    } else {
+      await this.deleteSession({ topic, emitEvent: false });
+    } else if (this.client.core.pairing.pairings.keys.includes(topic)) {
```

Comment:
Nice improvement 👍 

In retrospect I don't recall why we didn't combine pairing and session handling into this top level `disconnect`/why it was session-only and pairings had to be handled explicitly via `core.pairing.disconnect()`